### ISSUE TITLE
Question regarding the training loss

### ISSUE THREAD

**Issue by meneshail** [aqlaboratory/openfold#291]
Hi, thanks for presenting this great work. I am trying to train a model with a smaller dataset (~10000) from scratch, and have some questions about the loss scale and expected loss curve.

1. I notice the final loss will be multiplied by `sqrt(seq_len)` before backward, which is roughly a factor of 10~20 in most cases. So basically it's like using the original loss with a learning rate of 1e-2 or even larger (since the default lr is 1e-3).  Wouldn't it be a bit large and cause stability issues?

2. During the first 1000 steps of the training, I can hardly observe any decrease in my fape loss. So I was wondering if I have mistakenly used an inappropriate loss scale. If someone has successfully conducted the training and reached a fair accuracy, could you plot the curve of each loss term along the training steps, so that I could check with my loss scale and roughly get a sense of what loss curves I should expect?

My computation resource is still limited, so this kind of prior knowledge is really important to me. Any discussion or assistance is genuinely appreciated. Thanks in advance!

**Comment by lhatsk** [comment]
You should also monitor lDDT-CA. In my experience, FAPE loss is the last loss to decrease. Distogram and masked MSA loss have to reach a certain point first because the structure, in the beginning, is rubbish.

**Comment by meneshail** [comment]
Thanks for the reply! A step means a gradient descent step and I'm using a minibatch size of 48.
Yes, I think your explanation is reasonable. Basically, the learning of an informative embedding via masked MSA and distogram loss should precede a little the learning of the structure module.

May I ask, given this definition of step, when the fape loss starts to significantly decrease in your case? Also, are you using the default learning rate and scheduler setting? (1e-3 lr, warmup in the first 1000 steps)

**Comment by meneshail** [comment]
> You should also monitor lDDT-CA. In my experience, FAPE loss is the last loss to decrease. Distogram and masked MSA loss have to reach a certain point first because the structure, in the beginning, is rubbish and FAPE is a structure-related loss.
> 
> What is a step in your case?

Thanks for the reply! A step means a gradient descent step and I'm using a minibatch size of 48.
Yes, I think your explanation is reasonable. Basically, the learning of an informative embedding via masked MSA and distogram loss should precede a little the learning of the structure module. 

May I ask, given this definition of step, when the fape loss starts to significantly decrease in your case? Also, are you using the default learning rate and scheduler setting? (1e-3 lr, warmup in the first 1000 steps)

**Comment by lhatsk** [comment]
Sorry, I don't have this data atm, but it took some time. I don't think you see much FAPE movement before the distogram loss is < 0.5. Check out the OpenFold paper for some curves. lDDT-CA started to improve drastically around 3000 steps (Supplementary Figure 1). I didn't use warm-up.

**Comment by meneshail** [comment]
Sorry for my late reply. I indeed observed a drastic decrease in fape loss after 1000+ steps, along with a significant increase in lddt. I used 10000 proteins and the resulting curve highly resembled the one shown in Figure 6. Close the issue now.
